apps/api/src/mail/mail.service.ts

```diff
@@ -30,48 +29,43 @@ export class MailService {
       bodyType: 'html',
       body: mailOptions.body,
     };
-    const token = await this.getChesToken();
 
-    try {
-      const { data } = await axios.post<ChesResponse>(
-        `${process.env.CHES_SERVICE_HOST}/api/v1/email`,
-        emailBody,
-        {
-          headers: {
-            authorization: `Bearer ${token}`,
-            'content-type': 'application/json',
-          },
+    const token = await this.getChesToken();
+    const { data } = await axios.post<ChesResponse>(
+      `${process.env.CHES_SERVICE_HOST}/api/v1/email`,
+      emailBody,
+      {
+        headers: {
+          authorization: `Bearer ${token}`,
+          'content-type': 'application/json',
         },
-      );
-      return data;
-    } catch (e) {
-      throw new GenericException(MailError.FAILED_TO_SEND_EMAIL, e);
-    }
+        timeout: 20000,
+      },
+    );
+
+    return data;
   }
 
   /**
    * Auxiliary function to get access token from CHES
    *
    */
   private async getChesToken() {
-    try {
-      const token = await axios.post(
-        process.env.CHES_AUTH_URL as string,
-        new URLSearchParams({ grant_type: 'client_credentials' }),
-        {
-          auth: {
-            username: process.env.CHES_CLIENT_ID as string,
-            password: process.env.CHES_CLIENT_SECRET as string,
-          },
-          headers: {
-            'Content-Type': 'application/x-www-form-urlencoded',
-          },
+    const token = await axios.post(
+      process.env.CHES_AUTH_URL as string,
+      stringify({ grant_type: 'client_credentials' }),
```

Comment:
@vyasworks @hitankar-freshworks @arranfw @dbayly-freshworks 
some unknown issue is occurred regarding usage of `URLSearchParams` to construct request body. May be that is not well supported by Node.

Reply:
This is surprising, URLSearchParams looks fully supported and has worked well in the past, maybe we can revisit this implementation before opting for a library over native functionality

Reply:
`qs` is node js native lib

Reply:
Egg on my face!